boa/src/syntax/lexer/error.rs

```diff
@@ -14,7 +14,7 @@ impl From<io::Error> for Error {
 
 impl Error {
     /// Creates a new syntax error.
-    pub(super) fn syntax<M>(err: M) -> Self
+    pub(crate) fn syntax<M>(err: M) -> Self
```

Comment:
Why is this public at the crate level? I think we should only be creating lexer errors in the lexer.

Reply:
I agree, the syntax function is currently used elsewhere (outside crate) and so this change was needed

Reply:
I decided it was easier just to make it super for now and then go back and make the change later (starting at a baseline of code which compiles / can be tested)

Reply:
This is where it is used: https://github.com/boa-dev/boa/blob/496278c46025d2a5ce1a13a4012bcc2ce28ded04/boa/src/syntax/ast/token.rs#L117-L130 

I'm wondering if the token.rs file should actually be in the lexer directory rather than ast because the lexer is the one generating tokens.

Reply:
Commit f115892 shows the result of moving the token file